qutebrowser/completion/models/sortfilter.py

```diff
@@ -63,23 +61,18 @@ def set_pattern(self, val):
 
         Invalidates the filter and re-sorts the model.
 
-        If the current completion model overrides sort(), it is used.
-        If not, the default implementation in QCompletionFilterModel is called.
-
         Args:
             val: The value to set.
         """
         with debug.log_time(log.completion, 'Setting filter pattern'):
+            # empty value clears cache (not necessary for correctness, but
+            # helps with keeping memory requirements relatively low)
+            if not val:
+                self.srcmodel.filtered_out_cache = {}
             self.pattern = val
-            val = re.escape(val)
-            val = val.replace(r'\ ', r'.*')
-            self.pattern_re = re.compile(val, re.IGNORECASE)
             self.invalidateFilter()
             sortcol = 0
-            try:
-                self.srcmodel.sort(sortcol)
-            except NotImplementedError:
-                self.sort(sortcol)
+            self.sort(sortcol)
             self.invalidate()
```

Comment:
I tried to get rid of it in b5781f0ed3ad2fdd517afa37f7845919aa319c99 for #190 but it broke stuff with `completion -> shrink` according to the commit message of a6f1bf29aed0e041fa7e0c884d6166b4564a3908 (#204).

Unfortunately I don't remember the details, but some older related commits:

- c32ba2f2c898bc3bc8aa472a82fa7da9d18456d3
- a0bcfe19604361f564fac526c3a182ed0413faba
- 1f6ae3e0ed97c319dcff7e80f585af29a4566637
-